Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,996 @@
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+    }
+    else
+    {
+    # Fetch all front doors in the Subscription.
+        if ([String]::IsNullOrWhiteSpace($FilePath))
+        {
+            Write-Host "Fetching all Front Doors in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            # Get all Front Doors in a Subscription
+            $FrontDoorResources = Get-AzFrontDoor -ErrorAction Stop
+        }
+        else
+        {
+            if (-not (Test-Path -Path $FilePath))
+            {
+                Write-Host "Input file [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                return
+            }
+
+            Write-Host "Fetching all Front Doors from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $frontdoorDetails = Import-Csv -LiteralPath $FilePath
+            $validFrontDoorDetails = $frontdoorDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+            
+            $FrontDoorResources = $validFrontDoorDetails
+        }
+    }
+    $totalFrontDoors = ($FrontDoorResources | Measure-Object).Count
+
+    if ($totalFrontDoors -eq 0)
+    {
+        Write-Host "No Front Door found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        return
+    }
+  
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        Write-Host "Found [$($totalFrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else
+    {
+        Write-Host "Found [$($totalFrontDoors)] Front Door Endpoint(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)   
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+    
+    # Includes required version of TLS
+    $requiredMinTLSVersion = 1.2
+
+    # Includes Front Door Endpoints where Minimum required TLS version is not set to 1.2.
+    $nonCompliantEndpoints = @()
+
+    # Includes Front Door Endpoints where Minimum required TLS version is either set to 1.2 or blank.
+    $compliantEndpoints = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorEPsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
```

Comment:
NIT: From naming convention 'isMinTLSVersionSetOnCustomDomain' it looks like a boolean variable, but we are initializing it as empty array.

Please check if this is expected.

Reply:
Fixed.